crates/bitcoin-da/src/helpers/builders.rs

```diff
@@ -173,20 +173,18 @@ fn build_commit_transaction(
         None,
     );
 
-    // Find least one utxo from prev tx to order tx in one block
-    let (required_utxo, utxos) = if let Some(tx) = prev_tx {
-        let (requited_utxos, free_utxos) = utxos
-            .into_iter()
-            .partition::<Vec<_>, _>(|utxo| utxo.tx_id == tx.id);
```

Comment:
The thing is this one splits into prev_utxos and other utxos. It's possible that output of prev tx will be duplicated. Please check it.

Reply:
oof makes sense.